src/AppBundle/Procuration/Filter/ProcurationFilters.php

```diff
@@ -0,0 +1,215 @@
+        if ($page = $request->query->getInt(self::PARAMETER_PAGE, 1)) {
+            $filters->setCurrentPage($page);
+        }
+
+        return $filters;
+    }
+
+    public function setCurrentPage(int $page): void
+    {
+        if ($page < 1) {
+            $page = 1;
+        }
+
+        $this->currentPage = $page;
+    }
+
+    public function getCurrentPage(): int
+    {
+        return $this->currentPage;
+    }
+
+    public function getLimit(): int
+    {
+        return self::PER_PAGE;
+    }
+
+    public function setCountry(string $country): void
+    {
+        if (empty($country = strtoupper(trim($country)))) {
+            $this->country = null;
+
+            return;
+        }
+
+        if (!in_array($country, array_keys($this->getCountries()), true)) {
+            throw new ProcurationException(sprintf('Invalid country filter value given ("%s").', $country));
+        }
+
+        $this->country = trim($country);
+    }
+
+    public function getCountry(): ?string
+    {
+        return $this->country;
+    }
+
+    public function setCity(string $city): void
+    {
+        $this->city = trim($city);
+    }
+
+    public function getCity(): ?string
+    {
+        return $this->city;
+    }
+
+    public function setType(string $type): void
+    {
+        $this->type = isset(self::TYPES[$type]) ? $type : null;
+    }
+
+    public function setStatus(string $status): void
+    {
+        if (empty($status)) {
+            $this->status = null;
+
+            return;
+        }
+
+        $this->status = $status;
+    }
+
+    public function getStatus(): string
+    {
+        return $this->status;
+    }
+
+    public function getType(): ?string
+    {
+        return $this->type;
+    }
+
+    public function getTypes(): array
+    {
+        return self::TYPES;
+    }
+
+    public function getCountries(): array
+    {
+        return UnitedNationsBundle::getCountries();
+    }
+
+    public function hasData(): bool
+    {
+        return $this->country || $this->city || $this->type;
+    }
+
+    public function apply(QueryBuilder $qb, string $alias): void
+    {
+        if ($this->country) {
+            $qb->andWhere(sprintf('%s.voteCountry = :filterVotreCountry', $alias));
+            $qb->setParameter('filterVotreCountry', $this->country);
+        }
+
+        if ($this->city) {
+            if (is_numeric($this->city)) {
+                $qb->andWhere(sprintf('%s.votePostalCode LIKE :filterVoteCity', $alias));
+                $qb->setParameter('filterVoteCity', $this->city.'%');
+            } else {
+                $qb->andWhere(sprintf('LOWER(%s.voteCityName) LIKE :filterVoteCity', $alias));
+                $qb->setParameter('filterVoteCity', '%'.strtolower($this->city).'%');
+            }
+        }
+
+        if ($this->type) {
+            $qb->andWhere(sprintf('%s.%s = true', $alias, $this->type));
+        }
+
+        $qb
+            ->setFirstResult(($this->currentPage - 1) * $this->getLimit())
```

Comment:
IMHO the method `getLimit` is useless. You should use the constant directly here.